src/python/pants/backend/pants_info/explain_options.py

```diff
@@ -190,3 +220,32 @@ def console_output(self, targets):
                         inner_map["history"] = history_list
         if self.is_json():
             yield json.dumps(output_map, indent=2, sort_keys=True, cls=CoercingOptionEncoder)
+
+
+@goal_rule
+def explain_options(
+    explain_options_options: ExplainOptionsOptions,
+    options_wrapper: _Options,
+    global_options: GlobalOptions,
+    console: Console,
+) -> ExplainOptions:
+    eoo = explain_options_options.values
+    explainer = OptionsExplainer(
+        options_wrapper.options,
+        eoo.scope,
+        eoo.name,
+        eoo.rank,
+        eoo.show_history,
+        eoo.only_overridden,
+        eoo.skip_inherited,
+        eoo.output_format,
+        global_options.options.colors,
```

Comment:
Consider kwargs. Most of these params have the same type, so MyPy wouldn't be able to warn if we ever did something like reordered a param.